tools/sa-bench/main.go

```diff
@@ -0,0 +1,204 @@
+/*
```

Comment:
Collectd groups metrics by plugin.  A plugin will output multiple metrics in each json message that it sends.  For example, the network plugin will bundle tx_bytes, tx_packets, rx_bytes, etc.. into a single json message.  I am not sure that the code below has this structure.  

Also, the test sender needs to run continuously i.e -- default to running for 60 seconds but allow the user to set the runtime, -1 for forever (ctrl-c to break).  To keep things simple, you can just send one plugin per host and bundle all numMetrics metrics in that single message.

Reply:
I did try with collectd with AMQP1 plugin and I get the following outputs. 1 host contains several plugin as you mentioned, but also 1 plugin has several instance and each instance has several metric types, as well as type_instance. 
Collectd sends one combination of plugin/plugin_instance/type/type_instance in one message.

```
[{"values":[341],"dstypes":["derive"],"dsnames":["value"],"time":1517582264.007,"interval":10.000,"host":"f1237fb00214","plugin":"cpu","plugin_instance":"3","type":"cpu","type_instance":"nice"}]
[{"values":[11939],"dstypes":["derive"],"dsnames":["value"],"time":1517582264.007,"interval":10.000,"host":"f1237fb00214","plugin":"cpu","plugin_instance":"0","type":"cpu","type_instance":"interrupt"}]
```

Currently sa-bench only simulate simple plugin but I'd put above parameter in future feature.